### PR TITLE
Gnome-Shell 43 compatibility update.

### PR DESCRIPTION
**Pull request by glerroo**
Close issue: #749 and #751.

### CODE REVIEW

**Comment by GaryElshaw** [comment]
Unfortunately i'm still seeing this on Ubuntu: 'Tried to construct an object without a GType'.

File: system-monitor@paradoxxx.zero.gmail.com/extension.js [hunk 1]
```diff
@@ -1082,7 +1082,11 @@ const Battery = class SystemMonitor_Battery extends ElementBase {
         this.icon_hidden = false;
         this.percentage = 0;
         this.timeString = '-- ';
-        this._proxy = StatusArea.aggregateMenu._power._proxy;
+        if (shell_Version => '43') {
```
Comment (mgalgs) [inline system-monitor@paradoxxx.zero.gmail.com/extension.js:1085]: This is actually just defining an arrow function, which will always evaluate to true. I believe you want `shell_Version >= '43'`)

**Review by mgalgs — CHANGES_REQUESTED** [review]

**Review by mgalgs — COMMENTED** [review]
lgtm. Has this been tested on 43 as well as something before 43?

File: system-monitor@paradoxxx.zero.gmail.com/extension.js (hunk 1 above)
Comment (glerroo) [reply system-monitor@paradoxxx.zero.gmail.com/extension.js]: Yes you are right, i maked a typo, i have just corrected it.
Thanks

Comment (mgalgs) [reply system-monitor@paradoxxx.zero.gmail.com/extension.js]: :+1:

**Comment by mgalgs** [comment]
For anyone who may be interested, I've integrated these patches into `system-monitor-next`:

https://github.com/mgalgs/gnome-shell-system-monitor-applet/pull/26#issuecomment-1304926321

**Comment by averissimo** [comment]
Brilliant! I confirm that this is working on gnome 43.1 (Ubuntu 22.10 Kinetic)

**Comment by jens-br** [comment]
Thanks for this! :+1: Since this PR was created two months ago: Is there a rough time frame when it will be integrated? This would be very helpful!

Many thanks again!

**Comment by mgalgs** [comment]
@arderyp You can try https://extensions.gnome.org/extension/3010/system-monitor-next/, which is an extension built from [this fork](https://github.com/mgalgs/gnome-shell-system-monitor-applet) of this repo and contains this fix.

**Pull request merged**